src/DI/ContainerBuilder.php

```diff
@@ -348,11 +347,37 @@ public function prepareClassList()
 		}
 
 		//  build auto-wiring list
-		$this->classList = [];
+		$this->classList = $preferred = [];
 		foreach ($this->definitions as $name => $def) {
 			if ($class = $def->getImplement() ?: $def->getClass()) {
+				$defAutowired = $def->isAutowired();
+				if (is_array($defAutowired)) {
+					foreach ($defAutowired as $k => $aclass) {
+						if ($aclass === self::THIS_SERVICE) {
```

Comment:
Yes, but in what instances would you write `self` instead of writing directly the class name?
Also, it's not autowired when typehint === class name but when typehint instanceof class name, or not?